rustfmt-config/src/config_type.rs

```diff
@@ -365,12 +373,50 @@ macro_rules! create_config {
                     self.set().width_heuristics(WidthHeuristics::null());
                 }
             }
+
+            fn set_license_template(&mut self) {
+                let license_template_path = self.license_template_path();
+                let mut license_template_file = match File::open(&license_template_path) {
+                    Ok(file) => file,
+                    Err(e) => {
+                        eprintln!("Warning: unable to open license template file {:?}: {}",
+                                  license_template_path, e);
+                        return;
+                    }
+                };
+                let mut license_template_str = String::new();
+                match license_template_file.read_to_string(&mut license_template_str) {
+                    Ok(_) => (),
+                    Err(e) => {
+                        eprintln!("Warning: unable to read from license template file {:?}: {}",
+                                  license_template_path, e);
+                        return;
+                    }
+                }
+                let license_template_parsed = match parse_license_template(&license_template_str) {
+                    Ok(string) => string,
+                    Err(e) => {
+                        eprintln!("Warning: unable to parse license template file {:?}: {}",
+                                  license_template_path, e);
+                        return;
+                    }
+                };
+                self.license_template = match Regex::new(&license_template_parsed) {
+                    Ok(re) => Some(re),
+                    Err(e) => {
+                        eprintln!("Warning: regex syntax error in placeholder, unable to compile \
+                                   license template from file {:?}: {}", license_template_path, e);
+                        return;
+                    }
+                }
+            }
```

Comment:
This function would look nicer if you create a LicenseError enum with a display method for the messages and use the `?` operator.

Reply:
I agree it's ugly, I just thought I'd do the error handling here and avoid clutter at the call sites :) If I use `?`, how far up should I let the errors bubble? `set_license_template()` is called by functions which return unit, I suppose I should just pattern match there and let potential errors print?